content/en/platform/corda/5.3/reference/corda-cli/cpi.md

```diff
@@ -0,0 +1,81 @@
+---
+description: "List of CPI commands for the Corda CLI. Use these commands to upload or fetch CPIs."
+date: '2024-04-24'
+menu:
+  corda53:
+    identifier: corda53-cordacli-cpi
+    weight: 500
+    parent: corda53-cli-reference
+title: "cpi"
+---
+# cpi
+
+This section lists the Corda CLI `cpi` arguments. You can use these commands to upload CPIs to your Corda 5 cluster as well as list the CPIs that are already there.
+
+## upload
+
+The `upload` command uploads CPIs present in the specified `.cpi` file into your Corda 5 cluster, so that virtual nodes can be created using this CPI at a later point.
+
+<style>
+table th:first-of-type {
+    width: 30%;
+}
+table th:nth-of-type(2) {
+    width: 70%;
+}
+</style>
+
+| Argument           | Description                                                            |
+| ------------------ | ---------------------------------------------------------------------- |
+| -t, \-\-target     | The target address of the REST server, for example, https://host:port. |
+| -p, \-\-password   | REST password.                                                         |
+| -u, \-\-user       | REST user name.                                                        |
+| -c, \-\-cpi        | The CPI file to upload.                                                |
+| -w, \-\-wait       | Wait for the result or have a result ID returned to be checked later.  |
```

Comment:
Not sure I follow this description. Does specifying `-w` "wait" for the result AND return a result ID?

Reply:
@ShYogesh-R3 could you pls advise?

Reply:
Yes, `-w` wait for the result ID[checksum] to be populated. If we do not specify -w, it will return a normal `ID`, which we can use to get the the `checksum` eventually.

Reply:
Updated the description.